### ISSUE TITLE
[Core Feature] Flytekit client-side validation of pod specs

### ISSUE THREAD

**Issue by katrogan** [flyteorg/flyte#1123]
**Motivation: Why do you think this is important?**
Before serializing a pod spec, we should validate it. Although we use the kube client library to generate pod spec constructs, we should explore using the api client to validate [we can deserialize the pod spec](https://github.com/flyteorg/flytekit/blob/d52d4ff18e7db4d64674ba91c713fe1fcfd716e3/plugins/tests/pod/test_pod.py#L54,L57) before registering it.

**Goal: What should the final outcome look like, ideally?**
Faster turn-around for catching client-side pod spec validation issues.

**Describe alternatives you've considered**
The current solution of postponed debugging at execution time.

**[Optional] Propose: Link/Inline OR Additional context**
If you have ideas about the implementation please propose the change. If inline keep it short, if larger then you link to an external document.

**Comment by EngHabu** [comment]
Mind clarifying what additional checks this would give us? is it possible to generate an invalid pod spec if we are using kube client library?

**Comment by katrogan** [comment]
@EngHabu yes, there's been multiple cases on slack where someone entered an instance of something instead of a list (e.g. volume mounts) and deserialization at execution time failed confusingly
